src/ottplan.java

```diff
@@ -0,0 +1,141 @@
+import java.text.DecimalFormat;
+import java.util.*;
+abstract class Subscription{
+    protected String resolution;
+    protected String planName;
+    protected int noOfViewers;
+    protected double amount;
+    protected String dis;
+    abstract void calculate(boolean a);
```

Comment:
Don't include Discount logic inside the calculate logic. The discount got nothing to do with the subscription plan but it's only related to the user.